### PR TITLE
Ruby Path - Ruby on Rails -> SessionsCookiesAuthentication: Rewords a link

### PR DESCRIPTION
**Pull request by MLGabe**
…typo in another link

Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #XXXXX

**2. This PR:**
- Rewords the first link so it's easier to understand
- Fixes a small typo in another link

**3. Additional Information:**

### CODE REVIEW

File: ruby_on_rails/forms_and_authentication/sessions_cookies_authentication.md [hunk 1]
```diff
@@ -195,8 +195,8 @@ This section contains helpful links to other content. It isn't required, so cons
 ### Knowledge Check
 This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
 
-- <a class="knowledge-check-link" href="#using-cookies">How would you save a cookie to the browser if a users pet is a dog on their browser?</a>
+- <a class="knowledge-check-link" href="#using-cookies">How would you save a cookie on a user's browser, if their pet is a dog?</a>
 - <a class="knowledge-check-link" href="#login-check">How would you require a user is logged in _before_ running some code?</a>
 - <a class="knowledge-check-link" href="#http-authentication">Would you use Basic HTTP Authentication for authenticating users over alternatives such as the Devise gem?</a>
-- <a class="knowledge-check-link" href="#flash-message">How would you flash an error message on a users browser if they put a invalid username?</a>
+- <a class="knowledge-check-link" href="#flash-message">How would you flash an error message on a users browser if they put an invalid username?</a>
```
Comment (dm-murphy) [inline ruby_on_rails/forms_and_authentication/sessions_cookies_authentication.md:201]:
```suggestion
- <a class="knowledge-check-link" href="#flash-message">How would you flash an error message on a user's browser if they put an invalid username?</a>
```

**Review by dm-murphy — CHANGES_REQUESTED** [review]
Thanks for pointing this out MLGabe!

The first Knowledge Check is bizarre not only for the phrasing but also because there is no prior context for cookies using pets/dogs in this lesson. Additionally it is asking how a cookie would be saved, but that section of the lesson is actually talking about accessing the key-value pair of a cookie. 

I think that whole first Knowledge Check should be rewritten to reflect the section it is pointing back to and to remove the confusion about the pet/dog cookie. 

For the other Knowledge Check where you fixed the indefinite article, you just need an apostrophe for "user's browser" earlier in the line.

File: ruby_on_rails/forms_and_authentication/sessions_cookies_authentication.md (hunk 1 above)
Comment (MLGabe) [reply ruby_on_rails/forms_and_authentication/sessions_cookies_authentication.md:201]: > Additionally it is asking how a cookie would be saved, but that section of the lesson is actually talking about accessing the key-value pair of a cookie.

Wouldn't the following paragraph cover that?
`To work with cookies, Rails gives you access to a special hash called cookies, where each key-value pair is stored as a separate cookie on the user’s browser. If you were to save cookies[:hair-color] = "blonde", you’d be able to pull up your browser’s developer tools and see a cookie on the user’s browser that has a key of hair-color and a value of blonde. Delete it using cookies.delete(:hair-color).`

If so, the learner would probably answer the first question as follows:
cookies[:pet] = 'dog'

A possible rewrite, that would link back to that would be something along the lines:
>How would you retrieve from a cookie the user's hair color?

Whichever do you like best? 🤔

Comment (dm-murphy) [reply ruby_on_rails/forms_and_authentication/sessions_cookies_authentication.md:201]: What do you think of something like this:

`How do you set a cookie for hair color on a user's browser?`

That way we still focus on setting the cookie so a learner might answer with something like:
 `cookies[:hair-color] = 'black'`

**Review by dm-murphy — APPROVED** [review]
Thanks MLGabe!

**Pull request merged**